docker/DOCKER.md

```diff
@@ -0,0 +1,207 @@
+# Docker Images for balena CLI
+
+Docker images with balena CLI and docker-in-docker.
+
+## Features Overview
+
+These CLI images are based on the popular [Balena base images](https://www.balena.io/docs/reference/base-images/base-images/)
+so they include many of the features you see there.
+
+- Multiple Architectures:
+    - `rpi`
+    - `armv7hf`
+    - `aarch64` (debian only)
+    - `amd64`
+    - `i386`
+- Multiple Distributions
+    - `debian`
+    - `alpine`
+- [cross-build](https://www.balena.io/docs/reference/base-images/base-images/#building-arm-containers-on-x86-machines) functionality for building ARM containers on x86.
+- Helpful package installer script called `install_packages` inspired by [minideb](https://github.com/bitnami/minideb#why-use-minideb).
+
+## Image Names
+
+`balenalib/<arch>-<distro>-balenacli:<cli_ver>`
+
+- `<arch>` is the architecture and is mandatory. If using Dockerfile.templates, you can replace this with `%%BALENA_ARCH%%`.
+For a list of available device names and architectures, see the [Device types](https://www.balena.io/docs/reference/base-images/devicetypes/).
+- `<distro>` is the Linux distribution and is mandatory. Currently there are 2 distributions, namely `debian` and `alpine`.
+
+## Image Tags
+
+In the tags, all of the fields are optional, and if they are left out, they will default to their `latest` pointer.
```

Comment:
How about adding the following note here:
```
To ensure that the latest available CLI image is used, it is also necessary
to add the `--pull` flag to `docker run` commands. Otherwise, Docker will
reuse a locally cached image, without checking for available updates.
```
The CLI itself notifies users when new versions are available, but not less than 24 hours since installation, and not when executed as the `root` user. The last check timestamp is stored in `~/.config/configstore/update-notifier-balena-cli.json`. With these Docker images, I think users would never get notified of available updates. I think the CLI update notification mechanism will need improvements (in some other PR) to work better with these Docker images (and any installation where the CLI is executed as the root user).